paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java

```diff
@@ -277,35 +277,41 @@ private boolean buildForPostponeBucketCompaction(
 
             bucketOptions = new HashMap<>(table.options());
             bucketOptions.put(CoreOptions.BUCKET.key(), String.valueOf(bucketNum));
-            FileStoreTable realTable = table.copy(table.schema().copy(bucketOptions));
+            // bucket-postpone files won't be read if deletion vectors are set
```

Comment:
Maybe here should use `SnapshotReader`, needLookup will skip all level -2 and 0 files. There are two many magic in `TableScan`, we should use underly reader.

Please also add cases for other lookup cases.

Reply:
We use `Table` here, because currently all source operators rely on `ReadBuilder`, which is created from a `Table`.